From: "Mike Rapoport (Microsoft)" <rppt@kernel.org>
Subject: execmem: enforce allocation size aligment to PAGE_SIZE
Date: Wed, 23 Apr 2025 17:48:07 +0300

Before introduction of ROX cache execmem allocation size was always
implicitly aligned to PAGE_SIZE inside vmalloc.

However, when allocation happens from the ROX cache, this is not
enforced.

Make sure that the allocation size is always consistently aligned to
PAGE_SIZE.

Link: https://lkml.kernel.org/r/20250423144808.1619863-1-rppt@kernel.org
Fixes: 2e45474ab14f ("execmem: add support for cache of large ROX pages")
Signed-off-by: Mike Rapoport (Microsoft) <rppt@kernel.org>
Suggested-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Acked-by: Peter Zijlstra (Intel) <peterz@infradead.org>
Signed-off-by: Andrew Morton <akpm@linux-foundation.org>
---

 mm/execmem.c |    2 ++
 1 file changed, 2 insertions(+)

--- a/mm/execmem.c~execmem-enforce-allocation-size-aligment-to-page_size
+++ a/mm/execmem.c
@@ -377,6 +377,8 @@ void *execmem_alloc(enum execmem_type ty
 	pgprot_t pgprot = range->pgprot;
 	void *p;
 
+	size = PAGE_ALIGN(size);
+
 	if (use_cache)
 		p = execmem_cache_alloc(range, size);
 	else
_
